### PR TITLE
Implement interfaces for Timer and Duration

### PR DESCRIPTION
**Pull request by ancarda**
This commit pulls function signatures and DocBlocks into interfaces so downstream consumers are able to mock Timer and Duration in unit tests.

### CODE REVIEW

**Comment by sebastianbergmann** [comment]
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it.

**Comment by ancarda** [comment]
Could you please tell me why?

**Comment by sebastianbergmann** [comment]
`Duration` is an immutable value object, so there is no point to double it for testing. And an interface that is implemented by `Timer` also makes no sense as it is not required in the context of this component.

I understand that you use `Timer` in your code and would like to double it for testing. However, you should not double a type that you do not own. It is a common/best practice to define your own interface for something like `Timer`. Then you implement your interface in an adapter that uses the `Timer` class from this component for its implementation.

**TL;DR:** Do not directly use types from third-party libraries in your code. Hide them behind adapters that implement interfaces you own.

**Comment by ancarda** [comment]
Thank you @sebastianbergmann, that's given me a lot to think about.
